packages/core/docs/advanced/server-middleware.md

```diff
@@ -0,0 +1,138 @@
+# Server middleware
+
+The server middleware is a part of Vue Storefront networking. That's the way of making connections with the eCommerce platform and also a solution to reduce bundle size, number of calls, credentials sharing, extensibility handling, and more.
+
+## What is Vue Storefront Middleware and why we need it?
+
+The Vue Storefront middleware is an express.js proxy that taking the requests from the front-end translates them to a certain integration and call-related API-client.
+
+We have implemented it for a variety of reasons.
+
+First of all, it allows us to provide a prover way of extensibility - As a developer, you have control of the requests and the responses in the given platform and you can write an extension to add something extra in your project.
+
+A front-end application sometimes needs an additional API endpoint - that's also possible since you have a server written in express.js.
+
+All credentials for your platform are stored only on the backend side, there are no shared keys in the browser.
+
+## How it works (in a nutshell)
+
+The way of how it works represents the following diagram:
+
+<center>
+  <img src="../images/middleware-diagram.jpg" alt="Middleware Diagram" />
+</center>
+
+In the Vue Storefront platform, you always have a few components: Core, UI, Middleware, and integration part: composables, API-client (and sometimes UI). As we mentioned before, API-client is being called only on the middleware, but you still can access it on the front-end side - how is that possible?
+
+When you access an API-client on the front-end side, you are accessing actually a stub, instead of a real API-client instance. This stub makes a call to the middleware (Remote Procedure Call), and ask for loading a specific integration, and executing specific function.
```

Comment:
```suggestion
When you access an API-client on the front-end side, you are actually accessing a stub instead of the real API-client instance. This stub makes a call to the middleware (Remote Procedure Call) and requests a specific integration to execute a specific function.
```